src/privileged/blurts/FirefoxMonitor.jsm

```diff
@@ -1,926 +1,159 @@
+      this.startObserving();
       aExtension.callOnClose({
         close: () => {
-          stopObserving();
+          this.stopObserving();
         },
       });
     });
-
-    AddonManager.addAddonListener(this);
   },
 
-  onUninstalling(addon) {
-    this.handleDisableOrUninstall(addon);
-  },
-
-  onDisabled(addon) {
-    this.handleDisableOrUninstall(addon);
-  },
-
-  handleDisableOrUninstall(addon) {
-    if (addon.id !== gExtension.id) {
+  onStateChange(aBrowser, aWebProgress, aRequest, aStateFlags, aStatus) {
+    let location = aRequest.URI;
+    if (!aWebProgress.isTopLevel || aWebProgress.isLoadingDocument ||
+        !Components.isSuccessCode(aStatus)) {
       return;
     }
-    AddonManager.removeAddonListener(this);
-    // This is needed even for onUninstalling, because it nukes the addon
-    // from UI. If we don't do this, the user has a chance to "undo".
-    addon.uninstall();
-  },
-
-  eventListeners: new Set(),
-
-  addEventListener(aListener) {
-    this.eventListeners.add(aListener);
-  },
-
-  removeEventListener(aListener) {
-    this.eventListeners.delete(aListener);
-  },
-
-  notifyEventListeners(id) {
-    for (let cb of this.eventListeners) {
-      cb(id);
+    let host;
+    try {
+      host = Services.eTLD.getBaseDomain(location);
+    } catch (e) {
     }
+    if (!host) return;
+    this.warnIfNeeded(aBrowser, host);
   },
-};
-
-
-let observerAdded = false;
 
-function startObserving() {
-  const newtabURL = "about:newtab";
-
-  const tpl = {
-    onStateChange(aBrowser, aWebProgress, aRequest, aStateFlags, aStatus) {
-      let location = aRequest.URI;
-      if (!aWebProgress.isTopLevel || aWebProgress.isLoadingDocument ||
-          !Components.isSuccessCode(aStatus)) {
-        return;
-      }
-      let host;
-      try {
-        host = Services.eTLD.getBaseDomain(location);
-      } catch (e) {
-      }
-      if (!host) return;
-      warnIfNeeded(aBrowser, host);
-    },
-    onLocationChange(aBrowser, aWebProgress, aRequest, aLocation) {
-      if (!aWebProgress.isTopLevel || aLocation.spec !== newtabURL) {
-        return;
-      }
-      warnIfNeeded(aBrowser, newtabURL);
-    },
-  };
-
-  function tol(openEvent) {
-    const browser = openEvent.target.linkedBrowser;
-    if (browser.currentURI.spec !== newtabURL) {
+  onLocationChange(aBrowser, aWebProgress, aRequest, aLocation) {
+    if (!aWebProgress.isTopLevel || aLocation.spec !== this.kNewtabURL) {
       return;
     }
-    warnIfNeeded(browser, newtabURL);
-  }
+    this.warnIfNeeded(aBrowser, this.kNewtabURL);
+  },
 
-  EveryWindow.registerCallback(
-    "breach-alerts",
-    (win) => {
-      win.gBrowser.addTabsProgressListener(tpl);
-      win.gBrowser.tabContainer.addEventListener("TabOpen", tol);
-    },
-    (win) => {
-      if (!win.gBrowser) {
+  startObserving() {
+    const tol = (event) => {
+      const browser = event.target.linkedBrowser;
+      if (browser.currentURI.spec !== this.kNewtabURL) {
```

Comment:
Not sure how likely it is (since it seems to be working currently), but is it ever possible for `browser` to be undefined/null here?